### PR TITLE
For issue :Add test case for 2665: postscript fails and status=booted

### PR DESCRIPTION
**Pull request by junxiawang**
Add postscripts failed testcase
If postscripts failed the node status will return failed

### CODE REVIEW

**Comment by immarvin** [comment]
hi @junxiawang , please make sure you follow https://github.com/xcat2/xcat-process/blob/master/basic-process/The%20Template%20of%20Pull%20Request%20of%20Test%20Case.rst

File: xCAT-test/autotest/testcase/installation/reg_linux_diskfull_installation_flat_postscripts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+cmd:sleep 300
+cmd:a=0;while ! `lsdef -l $$CN|grep status|grep failed >/dev/null`; do sleep 20;((a++));if [ $a -gt 300 ];then break;fi done
+
+cmd:ping $$CN -c 3
+check:rc==0
+check:output=~64 bytes from $$CN
+cmd:lsdef -l $$CN | grep status
+check:rc==0
+check:output=~failed
+cmd:lsdef -l $$CN | grep updatestatus
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/installation/reg_linux_diskfull_installation_flat_postscripts]: why to check updatestatus?

File: xCAT-test/autotest/testcase/installation/reg_linux_diskless_installation_flat_postscripts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+cmd:sleep 900
+cmd:a=0;while ! `lsdef -l $$CN|grep status|grep failed >/dev/null`; do sleep 10;((a++));if [ $a -gt 60 ];then break;fi done
+
+cmd:ping $$CN -c 3
+check:rc==0
+check:output=~64 bytes from $$CN
+cmd:lsdef -l $$CN | grep status
+check:rc==0
+check:output=~failed
+cmd:lsdef -l $$CN | grep updatestatus
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/installation/reg_linux_diskless_installation_flat_postscripts]: same as above

**Comment by immarvin** [comment]
do we really need to perform an additional provision in this test case? it might introduce a lot of time cost

**Comment by junxiawang** [comment]
If add this case in our installation testcases there will be more check items so we could not probe the root cause if cases failed

**Comment by junxiawang** [comment]
Add 2  cases in this pull request.
These case(s) are added for issue:#2665
This pull request is for task :#2823

**Comment by whowutwut** [comment]
@junxiawang  You are able to modify your existing posts... there is a pencil icon to the right of the X. 
 I think you should add the issue numbers in the first post of this pull request so all the information is at the top.

File: xCAT-test/autotest/testcase/installation/reg_linux_diskfull_installation_flat_postscripts_failed [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+cmd:sleep 10
+cmd:a=0;while true; do [ $a -eq 100 ] && exit 1;output=$(makedhcp -q $$CN);[ $? -ne 0 ] && exit 1;echo $output|grep $$CN 2>/dev/null && exit 0;a=$[$a+1];sleep 1;done
+check:rc==0
+cmd:copycds $$ISO 
+check:rc==0
+cmd:/opt/xcat/share/xcat/tools/autotest/testcase/installation/setup_postscript_failed.sh
+check:rc==0
+cmd:chmod 777 /install/postscripts/test
+check:rc==0
+cmd:chdef $$CN  postscripts=test
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/installation/reg_linux_diskfull_installation_flat_postscripts_failed]: should this be ``chdef $$CN  -p postscripts=test``

File: xCAT-test/autotest/testcase/installation/reg_linux_diskless_installation_flat_postscripts_failed [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+cmd:a=0;while true; do [ $a -eq 100 ] && exit 1;output=$(makedhcp -q $$CN);[ $? -ne 0 ] && exit 1;echo $output|grep $$CN 2>/dev/null && exit 0;a=$[$a+1];sleep 1;done
+check:rc==0
+
+cmd:copycds $$ISO
+check:rc==0
+cmd:/opt/xcat/share/xcat/tools/autotest/testcase/installation/setup_postscript_failed.sh
+check:rc==0
+cmd:chmod 777 /install/postscripts/test
+check:rc==0
+cmd:chdef $$CN  postscripts=test
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/installation/reg_linux_diskless_installation_flat_postscripts_failed]: should this be ``cmd:chdef $$CN  -p postscripts=test``

**Comment by immarvin** [comment]
i am ok with the change

**Pull request merged**